app/assets/javascripts/components/PipelineSampleReport.jsx

```diff
@@ -17,6 +17,20 @@ class PipelineSampleReport extends React.Component {
     this.disableFilters = this.disableFilters.bind(this);
     this.enableFilters = this.enableFilters.bind(this);
   }
+  
+  componentDidMount() {
+    this.hideTags();
+    //only display download links when genus name is hovered on
+    $('.hover-wrapper').hover(function() {
+      $(this).find(".link-tag").show(); 
+    }, function() {
+      $(this).find(".link-tag").hide(); 
+    });
```

Comment:
This may be just me not understanding how browsers work, but it really looks like the browser is going to have to find all the elements with class .hover_wrapper and then for each one create a closure (a function) on the fly.

Why can't all the tags be hidden initially, with a hove function that shows them?   Wouldn't that let you eliminate the componentDidMount() function?

I am concerned about adding this function here because it represents a style of programming which doesn't generally scale for large pages, and will invite future additions of other code that is even less scalable.   All things equal, for this particular page, I think we would be well served to try and find a way of avoiding this.